src/Input/Select/Select.tsx

```diff
@@ -259,6 +270,17 @@ class Select extends React.Component<Props, State> {
       isLoading,
     } = this.state;
 
+    const deprecatedStatus = status || (error && 'error');
+
+    const shouldRenderError = !error || typeof error === 'boolean';
```

Comment:
Oh lines 275-282 seems a bit hard to grok for me :sweat_smile: 

I was confused by `shouldRenderError` because I assumed it was a flag to check if the user passed a function to `renderError`. But then we go on to check `renderError ? ` again. Maybe there is a more readable way to write this without the nested tenary? :pray:

Reply:
Yeah I see that this is confusing. I rewrote the logic to have a flag `shouldShowError` which actually determines whether we're going to render an error or not. And then I've replaced the nested ternary with a nested if/else, not beautiful but more readable. What do you think?

Reply:
Yeap, more readable. Thanks!